### PR TITLE
feat:added authentication functionality

### PR DESCRIPTION
**Pull request by mariyasada**
-added authentication(login,signup,logout)
-added address management static page
-added toast for login and logout
-added logout page

https://user-images.githubusercontent.com/77275100/163443762-27356b26-badf-413f-b9b8-3fbc413f9ff8.mp4

### CODE REVIEW

File: src/components/nav/Nav.jsx [hunk 1]
```diff
@@ -56,11 +62,18 @@ export const Nav = () => {
       <div className="nav-social-media-icons-container">
         <ul className="nav-items">
           <li className="nav-item">
-            <NavLink to="/login">
-              <span className="nav-icon">
-                <FaUserAlt title="Log In" />
-              </span>
-            </NavLink>
+            <span className="nav-icon">
+              {console.log(isloggedIn)}
```
Comment (tusharkandpal) [inline src/components/nav/Nav.jsx]: remove this console.log please

File: src/pages/SignUp/signUp.jsx [hunk 2]
```diff
@@ -1,109 +1,156 @@
...
-import { FaEyeSlash } from "react-icons/fa";
+import { FaEyeSlash, FaEye } from "react-icons/fa";
 import { BsExclamationTriangle } from "react-icons/bs";
+import { initialSignUpData } from "../../constants/auth-constants";
+import { useAuth } from "../../context/Authentication/auth-context";
 
 export const SignUpPage = () => {
+  const [signupData, setSignUpData] = useState(initialSignUpData);
+  const [passVisible, setPassVisible] = useState(true);
+  const [conPassVisible, setConPassVisible] = useState(true);
```
Comment (tusharkandpal) [inline src/pages/SignUp/signUp.jsx]: you can name them better.

File: src/pages/UserProfilepage/component/User.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+
+export const User = () => {
+  return (
+    <>
+      {/* <div className="user-detail details" id="profile"> */}
```
Comment (tusharkandpal) [inline src/pages/UserProfilepage/component/User.jsx]: remove this logger

File: src/pages/UserProfilepage/component/User.jsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        <div className="fullname-detail">
+          <span className="text-size-md">Full Name : </span>
+          <span className="text-size-md">Admin </span>
+        </div>
+        <div className="email-detail text-size-md">
+          <span>Email :</span>
+          <span> test@gmail.com </span>
+        </div>
+      </div>
+      {/* </div> */}
```
Comment (tusharkandpal) [inline src/pages/UserProfilepage/component/User.jsx]: this too

**Review by tusharkandpal — APPROVED** [review]
well implemented functionality.

suggestion: naming variables, or states can be improved.

File: src/components/nav/Nav.jsx [hunk 5]
```diff
@@ -1,17 +1,23 @@
...
 import { useCartandWishList } from "../../context/CartAndWishlist-context";
+import { useAuth } from "../../context/Authentication/auth-context";
 
 export const Nav = () => {
   const { state, dispatch } = useProduct();
   const { cartState, wishListState } = useCartandWishList();
   const { cartItem, cartCount } = cartState;
   const { wishListItem, wishListCount } = wishListState;
+  const { user } = useAuth();
+  const { isloggedIn } = user;
```
Comment (dev-enforced) [inline src/components/nav/Nav.jsx]: You can destructure it in a single line as 
```
const {user:{isLoggedIn}}=useAuth();
```
if it doesn't work you can continue in this manner too!!

File: src/context/Authentication/auth-context.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import { createContext,useContext,useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { signupService,logInService,signOutService } from "../../Services/Authservice/index";
```
Comment (dev-enforced) [inline src/context/Authentication/auth-context.js]: No need to mention index in the end . The imports works without it too
```
import {services which you have} from ''../../Services/Authservice"
```

**Review by dev-enforced — APPROVED** [review]
Good work on the authentication feature @mariyasada .Liked the toasts functionality. Do apply the pointers suggested by @tusharkandpal to remove the logger messages and improving variable names. Gave some minor suggestions which you can add on later.!!

File: src/pages/SignUp/signUp.jsx (hunk 2 above)
Comment (mariyasada) [reply src/pages/SignUp/signUp.jsx]: okay will do

**Pull request merged**